log improvements + deflake test + add missing test

knz opened the pull request:
### util/log: make SIGUSR1 and SIGUSR2 flush and rotate the logs

It is common while developing system administration tools and in tests
to need to flush the log files and rotate them. Prior to this patch,
there was no way to do this in CockroachDB short of restarting the
server.

Release note (general change): a CockroachDB process will now flush
its logs upon receiving `SIGUSR1` and rotate the log files upon
receiving `SIGUSR2`. This is aimed to simplify the automation of
process monitoring, test and backup tools.

###  cli/interactive_tests: deflake the audit log test

Prior to this patch, the test required a server restart to rotate the
logs. This in turn could cause the SQL lease on the table to not be
properly released, which would cause the test to hang after the
restart.

This patch fixes that by using a signal to rotate the logs instead.

Fixes #23714.

### sql: add the missing tests for the execution log

I forgot to add the tests for the exec log to my previous commit "sql:
enhance statement logging".

Here they are.

knz committed:
sql: add the missing tests for the execution log

I forgot to add the tests for the exec log to my previous commit "sql:
enhance statement logging".

Here they are.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/23760)
<!-- Reviewable:end -->

andreimatei commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

*[pkg/cli/interactive_tests/test_exec_log.tcl, line 40 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23760#-L7R9p8uEO5d2-3j9gdF:-L7R9p8uEO5d2-3j9gdG:b5zwnt6) ([raw file](https://github.com/cockroachdb/cockroach/blob/1284cf854479923d155b1820a2bdb21cd36baf62/pkg/cli/interactive_tests/test_exec_log.tcl#L40)):*
> ```Tcl
> 
> # We can't flush the log externally, so shut down the server to force a flush.
> interrupt
> ```

ugh. Have you considered adding a signal handler or a `/debug` endpoint that flushes the logs? I would have used it in the past if it existed. Maybe it's easy...

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23760#-:-L7RA27F-gSt1QnDSzbs:bnfp4nl)*
<!-- Sent from Reviewable.io -->

knz committed:
util/log: make SIGUSR1 and SIGUSR2 flush and rotate the logs

It is common while developing system administration tools and in tests
to need to flush the log files and rotate them. Prior to this patch,
there was no way to do this in CockroachDB short of restarting the
server.

Release note (general change): a CockroachDB process will now flush
its logs upon receiving `SIGUSR1` and rotate the log files upon
receiving `SIGUSR2`. This is aimed to simplify the automation of
process monitoring, test and backup tools.

knz committed:
cli/interactive_tests: deflake the audit log test

Prior to this patch, the test required a server restart to rotate the
logs. This in turn could cause the SQL lease on the table to not be
properly released, which would cause the test to hang after the
restart.

This patch fixes that by using a signal to rotate the logs instead.

Release note: None

knz commented:
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/cli/interactive_tests/test_exec_log.tcl, line 40 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23760#-L7R9p8uEO5d2-3j9gdF:-L7UD6_36RttkJFFlUc8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1284cf854479923d155b1820a2bdb21cd36baf62/pkg/cli/interactive_tests/test_exec_log.tcl#L40)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

ugh. Have you considered adding a signal handler or a `/debug` endpoint that flushes the logs? I would have used it in the past if it existed. Maybe it's easy...
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/23760)*
<!-- Sent from Reviewable.io -->
